src/persist/upcaster.rs

```diff
@@ -62,18 +62,10 @@ impl FromStr for SemanticVersion {
     type Err = SemanticVersionError;
 
     fn from_str(event_version: &str) -> Result<Self, Self::Err> {
-        let split_version: Vec<&str> = event_version.split('.').collect();
-        let patch = if split_version.len() < 3 {
-            0
-        } else {
-            u32::from_str(split_version.get(2).unwrap())?
-        };
-        let minor_version = if split_version.len() < 2 {
-            0
-        } else {
-            u32::from_str(split_version.get(1).unwrap())?
-        };
-        let major_version = u32::from_str(split_version.first().unwrap())?;
+        let mut split_version = event_version.split('.').fuse();
+        let major_version = u32::from_str(split_version.next().unwrap())?;
```

Comment:
You've removed the length check so we can't safely `unwrap` here.

Reply:
no, i think this is ok- even if the 'event_version' is an empty string, `event_version.split('.').next()` will evaluate to `Some("")`, so the unwrap is sound.

You can see that from one of the examples in the docs for the `split` method-

```rust
let v: Vec<&str> = "".split('X').collect();
assert_eq!(v, [""]);
```

Of course if the string is empty, `u32::from_str` will fail

Reply:
Very good, thank you.